src/server/scripts/EasternKingdoms/Stratholme/boss_baron_rivendare.cpp

```diff
@@ -106,7 +106,8 @@ class boss_baron_rivendare : public CreatureScript
         void AttackStart(Unit* who) override
         {
             //can't use entercombat(), boss' dmg aura sets near players in combat, before entering the room's door
-            instance->SetData(TYPE_BARON, IN_PROGRESS);
+            if (instance->GetData(TYPE_BARON) == NOT_STARTED)
```

Comment:
I don't get that check. What makes IN_PROGRESS be set while it's set to anything else but NOT_STARTED?